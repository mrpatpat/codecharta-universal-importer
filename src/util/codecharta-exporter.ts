import { AnalyzedFile } from "../model/analyzed-file";

export class CodechartaExporter {

    public static build(projectName: string, analyzedFiles: AnalyzedFile[]) :string {

        const cc = {
            "projectName": projectName,
            "apiVersion": "1.0",
            "nodes": [
                {
                    "name": "root",
                    "type": "Folder",
                    "children": []
                }
            ]
        };

        return JSON.stringify(cc);

    }

}