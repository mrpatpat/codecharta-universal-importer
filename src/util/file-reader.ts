import { readdirSync, lstatSync, readFileSync } from "fs-extra";
import * as path from "path";

export class FileReader {

    public collectFiles(dir: string): string[] {
        let result: string[] = [];
        readdirSync(dir).forEach(file => {
            const fullPath = path.join(dir, file);
            if (lstatSync(fullPath).isDirectory()) {
                result = result.concat(this.collectFiles(fullPath));
            } else {
                result.push(fullPath.replace(/\\/g,"/"));
            }
        });
        return result;
    }

    public readFile(path: string): string {
        return readFileSync(path, 'utf8');
    }

}