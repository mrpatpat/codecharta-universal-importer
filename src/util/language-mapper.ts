import { Languages } from "../model/languages";

export class LanguageMapper {

    public static getLanguageFromPath(path: string) {
        const suffix = path.split(".").pop();
        switch(suffix) {
            case "ts": 
            case "tsx":
                return Languages.TYPESCRIPT;
            default:     
                return Languages.UNKNOWN;
        };
    }

}