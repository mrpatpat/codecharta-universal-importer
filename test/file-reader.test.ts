import { FileReader } from "../src/util/file-reader"

describe("FileReader", () => {

  it("should read multiple files in a given directory", async () => {
    const fr = new FileReader();
    const result = fr.collectFiles("test/sample-projects");
    expect(result).toEqual([
      "test/sample-projects/typescript/hello-universe.ts",
      "test/sample-projects/typescript/hello-world.ts"
    ]);
  })

  it("should read multiple file contents in a given directory", async () => {
    const fr = new FileReader();
    const files = fr.collectFiles("test/sample-projects");
    const contents = files.map(file => fr.readFile(file));
    expect(contents).toMatchSnapshot();
  })

})
