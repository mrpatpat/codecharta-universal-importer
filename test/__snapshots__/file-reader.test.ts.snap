// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileReader should read multiple file contents in a given directory 1`] = `
Array [
  "export class HelloUniverse {

    public getHello(): string {
        return \\"Hello Universe!\\";
    }

    public sayHello(): void {
        console.log(this.getHello());
    }

}",
  "export class HelloWorld {

    public getHello(): string {
        return \\"Hello World!\\";
    }

    public sayHello(): void {
        console.log(this.getHello());
    }

}",
]
`;
